portal-impl/src/com/liferay/portal/model/impl/UserImpl.java

```diff
@@ -93,11 +93,7 @@ public class UserImpl extends UserBaseImpl {
 
 	@Override
 	public Contact fetchContact() {
-		if (_contact == _NULL_CONTACT) {
-			return null;
-		}
-
-		if (_contact == null) {
+		if ((_contact == null) || (_contact == _NULL_CONTACT)) {
```

Comment:
@brianikim , I don't get this: before if the _contact was already "calculated" we returned null.
Now we will do the fetch every time

Reply:
To further clarify, the _contact was never "calculated" in the fetch call, it was always returning null beacause `_contact == new ContactImpl()`. If `getContact()` was never called prior to `fetchContact()`, `fetchContact()` always returns null for wrong reasons.